joplin/pages/service_page/tests.py

```diff
@@ -131,3 +112,74 @@ def test_create_service_page_with_dynamic_content_list():
 def test_kitchen_sink():
     page = fixtures.kitchen_sink()
     assert isinstance(page, ServicePage)
+
+
+@pytest.mark.django_db
+def test_create_service_page_with_step_with_1_location():
+    page = fixtures.step_with_1_location()
+    expected_steps = components.step_with_1_location()
+
+    assert isinstance(page, ServicePage)
+    assert page.title == "Service Page with 1 location step"
+    assert page.slug == "service-page-with-1-location-step"
+    for i, step in enumerate(page.steps.stream_data):
+        assert step["type"] == expected_steps[i]["type"]
+        assert step["value"] == expected_steps[i]["value"]
+
+
+@pytest.mark.django_db
+def test_create_service_page_with_step_with_2_locations():
+    page = fixtures.step_with_2_locations()
+    expected_steps = components.step_with_2_locations()
+
+    assert isinstance(page, ServicePage)
+    assert page.title == "Service Page with 2 locations step"
+    assert page.slug == "service-page-with-2-locations-step"
+    for i, step in enumerate(page.steps.stream_data):
+        assert step["type"] == expected_steps[i]["type"]
+        assert step["value"] == expected_steps[i]["value"]
+
+
+@pytest.mark.django_db
+def test_import_step_with_1_already_existing_location(remote_staging_preview_url, test_api_url, test_api_jwt_token):
+    location_page = location_page_fixtures.title()
+    url = f'{remote_staging_preview_url}/services/UGFnZVJldmlzaW9uTm9kZTo1OA==?CMS_API={test_api_url}'
+    page = PageImporter(url, test_api_jwt_token).fetch_page_data().create_page()
+    assert isinstance(page, ServicePage)
+    assert page.live
+    # The importer should assign the step's location to our matching location_page.
+    assert page.steps.stream_data[0]["value"]["locations"][0] == location_page.pk
+
+
+@pytest.mark.django_db
+def test_import_step_with_1_new_location(remote_staging_preview_url, test_api_url, test_api_jwt_token):
+    url = f'{remote_staging_preview_url}/services/UGFnZVJldmlzaW9uTm9kZTo1OA==?CMS_API={test_api_url}'
+    page = PageImporter(url, test_api_jwt_token).fetch_page_data().create_page()
+    assert isinstance(page, ServicePage)
+    assert not page.live
+    # The importer should delete the location because it hasn't been imported yet.
+    # TODO: update API to allow creation of location_page from step data
+    assert not page.steps
+
+
+@pytest.mark.django_db
+def test_import_step_with_1_already_existing_location_1_new_location(remote_staging_preview_url, test_api_url, test_api_jwt_token):
+    location_page = location_page_fixtures.title()
+    url = f'{remote_staging_preview_url}/services/UGFnZVJldmlzaW9uTm9kZTo2Mg==?CMS_API={test_api_url}'
+    page = PageImporter(url, test_api_jwt_token).fetch_page_data().create_page()
+    assert isinstance(page, ServicePage)
+    assert not page.live
+    assert len(page.steps.stream_data[0]["value"]["locations"]) == 1
```

Comment:
might be nice to have a comment in here explaining why we want it to be 1
`# The importer should delete one of the two locations because it hasn't been imported yet.`